lib/plug/conn.ex

```diff
@@ -220,7 +223,8 @@ defmodule Plug.Conn do
             script_name: [],
             secret_key_base: nil,
             state: :unset,
-            status: nil
+            status: nil,
+            version: nil
```

Comment:
Are we really sure the version needs to be in the conn? It will be a very infrequently accessed value and having it here adds a one word overhead to every conn mutation.

I think a simple function (with an additional adapter callback) would be fine.